src/index.js

```diff
@@ -25,23 +25,66 @@ const { fetch, Response } = process.env.HELIX_FETCH_FORCE_HTTP1
   : fetchAPI;
 const commence = require('./stop');
 
+function getMdUrl(host, path, log) {
+  let mdPath;
+  const file = path.split('/').pop() || 'index'; // use 'index' if no filename
+  if (file.endsWith('.html')) {
+    mdPath = path.replace(/\.html$/, '.md');
+  } else if (file && !file.includes('.')) {
+    mdPath = `${path}.md`;
+  }
+  if (!mdPath) {
+    log.debug('not an html document, so no markdown purging required');
+    return null;
+  }
+  const ghDetails = host.split('.')[0].split('--');
+  if (ghDetails.length < 2) {
+    log.warn('invalid inner cdn url');
+    return null;
+  }
+  const owner = ghDetails.pop();
+  const repo = ghDetails.pop();
+  const branch = ghDetails[0] || 'master';
+  return `https://${branch}--${repo}--${owner}.hlx.page${mdPath}`;
```

Comment:
this is actually not correct - we should (also) purge the current `sha`
see https://github.com/adobe/helix-pipeline/issues/985

Reply:
As communicated in https://github.com/adobe/helix-pipeline/issues/985#issuecomment-788233912, we will stick with the branch approach.